exercises/matrix/src/test/java/MatrixTest.java

```diff
@@ -0,0 +1,184 @@
+
+@RunWith(Enclosed.class)
+public class MatrixTest {
+
+    @RunWith(Parameterized.class)
+    public static class GetFirstRowTest {
+        private Matrix matrix;
+        private int[] firstRow;
+
+        @Parameterized.Parameters(name = "{index}: checking first row of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0, 1}},
+                    {"0\\n1", new int[] {0}},
+                    {"0 1\\n2 3", new int[] {0, 1}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {0, 1, 2}}
+            });
+        }
+
+        public GetFirstRowTest(String matrixAsString, int[] firstRow) {
+            this.matrix = new Matrix(matrixAsString.replaceAll("\\\\n", "\n"));
+            this.firstRow = firstRow;
+        }
+
+        @Test
+        public void getFirstRowTest() {
+            assertArrayEquals(matrix.getRow(0), firstRow);
+        }
+    }
+
+    @RunWith(Parameterized.class)
+    public static class GetLastRowTest {
+        private Matrix matrix;
+        private int[] lastRow;
+
+        @Parameterized.Parameters(name = "{index}: checking last row of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0, 1}},
+                    {"0\\n1", new int[] {1}},
+                    {"0 1\\n2 3", new int[] {2, 3}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {6, 7, 8}}
+            });
+        }
+
+        public GetLastRowTest(String matrixAsString, int[] lastRow) {
+            this.matrix = new Matrix(matrixAsString.replaceAll("\\\\n", "\n"));
+            this.lastRow = lastRow;
+        }
+
+        @Ignore
+        @Test
+        public void getLastRowTest() {
+            assertArrayEquals(matrix.getRow(matrix.getRowsCount() - 1), lastRow);
+        }
+    }
+
+    @RunWith(Parameterized.class)
+    public static class GetFirstColumnTest {
+        private Matrix matrix;
+        private int[] firstColumn;
+
+        @Parameterized.Parameters(name = "{index}: checking first column of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0}},
+                    {"0\\n1", new int[] {0, 1}},
+                    {"0 1\\n2 3", new int[] {0, 2}},
+                    {"0 1 2\\n3 4 5\\n6 7 8", new int[] {0, 3, 6}}
+            });
+        }
+
+        public GetFirstColumnTest(String matrixAsString, int[] firstColumn) {
+            this.matrix = new Matrix(matrixAsString.replaceAll("\\\\n", "\n"));
+            this.firstColumn = firstColumn;
+        }
+
+        @Ignore
```

Comment:
I think these `@Ignore`s could also be applied so the inner classes themselves instead of the nested `@Test` methods, which could make them a little easier to spot? e.g.

```java
@Ignore
@RunWith(Parameterized.class)
public static class GetLastRowTest {
```

As before, though, it might just be me that finds the top-level annotation easier to see!